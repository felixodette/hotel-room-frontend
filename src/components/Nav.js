/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { FaBars, FaTwitter } from 'react-icons/fa';
import { GrFacebookOption } from 'react-icons/gr';
import { AiFillLinkedin } from 'react-icons/ai';

const Nav = () => (
  <>
    <nav id="sidebarMenu" className="collapse d-lg-block sidebar">
      <div className="position-sticky">
        <div className="list-group list-group-flush mx-3 mt-4">

          <a href="/" className="list-group-item list-group-item-action py-2 ripple active" aria-current="true">
            <span>Home</span>
          </a>
          <a href="/room-new" className="list-group-item list-group-item-action py-2 ripple ">
            <span>Add Room</span>
          </a>
          <a href="/room-delete" className="list-group-item list-group-item-action py-2 ripple">
            <span>Delete Room</span>
          </a>
          <a href="/reservations-new" className="list-group-item list-group-item-action py-2 ripple">
            <span>Make Reservations</span>
          </a>
          <a href="/reservations" className="list-group-item list-group-item-action py-2 ripple">
            <span>My Reservations</span>
          </a>
          <a href="/user" className="list-group-item list-group-item-action py-2 ripple">
            <span>Login</span>
          </a>
        </div>

      </div>
      <div className="card-footer bg-transparent border-0 fixed-bottom w-25">
        <GrFacebookOption
          onClick={() => {
            window.open(`https://www.facebook.com/sharer/sharer.php?u=${window.location.href}`, '_blank');
          }}
          className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2 mx-2 social-icons"
        />
        <FaTwitter
          onClick={() => {
            window.open(`https://twitter.com/intent/tweet?url=${window.location.href}`, '_blank');
          }}
          className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2 mx-2 social-icons"
        />
        <AiFillLinkedin
          onClick={() => {
            window.open(`https://www.linkedin.com/shareArticle?mini=true&url=${window.location.href}`, '_blank');
          }}
          className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle mx-2 p-2 social-icons"
        />
      </div>
    </nav>

    <nav id="main-navbar" className="navbar navbar-expand-lg fixed-top">

      <div className="container-fluid">

        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#sidebarMenu"
          aria-controls="sidebarMenu"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <FaBars />
        </button>

        <a className="navbar-brand" href="/">
          <img
            src="https://res.cloudinary.com/wonderwoman/image/upload/v1665273260/3D_gold_crown_logo_-_Made_with_PosterMyWall_n1vleo.png"
            alt="Logo"
            loading="lazy"
          />
        </a>
      </div>

    </nav>
  </>
);

export default Nav;
