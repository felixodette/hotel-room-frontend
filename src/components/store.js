import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../redux/reducers/index';

const initialState = {};
const middleware = [thunk];
const store = createStore(
  rootReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
/* eslint-disable */
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()));
/* eslint-enable */

export default store;
