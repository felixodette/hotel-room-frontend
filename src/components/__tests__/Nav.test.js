import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '../Nav';
import store from '../../redux/configureStore';

describe('Sidebar Navigation', () => {
  render(
    <Provider store={store}>
      <Router>
        <Nav />
      </Router>
    </Provider>,
  );

  test('Test the dom for successful rendering of elements', () => {
    const Home = screen.getByText('Home');
    expect(Home).toBeInTheDocument();
  });
});
