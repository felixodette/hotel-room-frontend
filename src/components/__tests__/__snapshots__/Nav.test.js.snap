// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Navigation It correctly renders a nav component 1`] = `
<div>
  <nav
    class="collapse d-lg-block sidebar"
    id="sidebarMenu"
  >
    <div
      class="position-sticky"
    >
      <div
        class="list-group list-group-flush mx-3 mt-4"
      >
        <a
          aria-current="true"
          class="list-group-item list-group-item-action py-2 ripple active"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
        <a
          class="list-group-item list-group-item-action py-2 ripple "
          href="/room-new"
        >
          <span>
            Add Room
          </span>
        </a>
        <a
          class="list-group-item list-group-item-action py-2 ripple"
          href="/room-delete"
        >
          <span>
            Delete Room
          </span>
        </a>
        <a
          class="list-group-item list-group-item-action py-2 ripple"
          href="/reservations-new"
        >
          <span>
            Make Reservations
          </span>
        </a>
        <a
          class="list-group-item list-group-item-action py-2 ripple"
          href="/reservations"
        >
          <span>
            My Reservations
          </span>
        </a>
        <a
          class="list-group-item list-group-item-action py-2 ripple"
          href="/user"
        >
          <span>
            Login
          </span>
        </a>
      </div>
    </div>
    <div
      class="card-footer bg-transparent border-0 fixed-bottom w-25 socials"
    >
      <svg
        class="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2 mx-2 social-icons"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.94474914,22 L9.94474914,13.1657526 L7,13.1657526 L7,9.48481614 L9.94474914,9.48481614 L9.94474914,6.54006699 C9.94474914,3.49740494 11.8713513,2 14.5856738,2 C15.8857805,2 17.0033128,2.09717672 17.3287076,2.13987558 L17.3287076,5.32020466 L15.4462767,5.32094085 C13.9702212,5.32094085 13.6256856,6.02252733 13.6256856,7.05171716 L13.6256856,9.48481614 L17.306622,9.48481614 L16.5704347,13.1657526 L13.6256856,13.1657526 L13.6845806,22"
          fill-rule="evenodd"
        />
      </svg>
      <svg
        class="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2 mx-2 social-icons"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
        />
      </svg>
      <svg
        class="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle mx-2 p-2 social-icons"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
        />
      </svg>
    </div>
  </nav>
  <nav
    class="navbar navbar-expand-lg fixed-top"
    id="main-navbar"
  >
    <div
      class="container-fluid"
    >
      <button
        aria-controls="sidebarMenu"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#sidebarMenu"
        data-bs-toggle="collapse"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </button>
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt="Logo"
          loading="lazy"
          src="https://res.cloudinary.com/wonderwoman/image/upload/v1665273260/3D_gold_crown_logo_-_Made_with_PosterMyWall_n1vleo.png"
        />
      </a>
    </div>
  </nav>
</div>
`;

exports[`Sidebar Navigation It renders the Nav correctly 1`] = `
Array [
  <nav
    className="collapse d-lg-block sidebar"
    id="sidebarMenu"
  >
    <div
      className="position-sticky"
    >
      <div
        className="list-group list-group-flush mx-3 mt-4"
      >
        <a
          aria-current="true"
          className="list-group-item list-group-item-action py-2 ripple active"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
        <a
          className="list-group-item list-group-item-action py-2 ripple "
          href="/room-new"
        >
          <span>
            Add Room
          </span>
        </a>
        <a
          className="list-group-item list-group-item-action py-2 ripple"
          href="/room-delete"
        >
          <span>
            Delete Room
          </span>
        </a>
        <a
          className="list-group-item list-group-item-action py-2 ripple"
          href="/reservations-new"
        >
          <span>
            Make Reservations
          </span>
        </a>
        <a
          className="list-group-item list-group-item-action py-2 ripple"
          href="/reservations"
        >
          <span>
            My Reservations
          </span>
        </a>
        <a
          className="list-group-item list-group-item-action py-2 ripple"
          href="/user"
        >
          <span>
            Login
          </span>
        </a>
      </div>
    </div>
    <div
      className="card-footer bg-transparent border-0 fixed-bottom w-25 socials"
    >
      <svg
        className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2 mx-2 social-icons"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.94474914,22 L9.94474914,13.1657526 L7,13.1657526 L7,9.48481614 L9.94474914,9.48481614 L9.94474914,6.54006699 C9.94474914,3.49740494 11.8713513,2 14.5856738,2 C15.8857805,2 17.0033128,2.09717672 17.3287076,2.13987558 L17.3287076,5.32020466 L15.4462767,5.32094085 C13.9702212,5.32094085 13.6256856,6.02252733 13.6256856,7.05171716 L13.6256856,9.48481614 L17.306622,9.48481614 L16.5704347,13.1657526 L13.6256856,13.1657526 L13.6845806,22"
          fillRule="evenodd"
        />
      </svg>
      <svg
        className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle p-2 mx-2 social-icons"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
        />
      </svg>
      <svg
        className="fs-1 opacity-50 mb-3 mb-md-4 border border-2 border-dark rounded-circle mx-2 p-2 social-icons"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
        />
      </svg>
    </div>
  </nav>,
  <nav
    className="navbar navbar-expand-lg fixed-top"
    id="main-navbar"
  >
    <div
      className="container-fluid"
    >
      <button
        aria-controls="sidebarMenu"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#sidebarMenu"
        data-bs-toggle="collapse"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </button>
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt="Logo"
          loading="lazy"
          src="https://res.cloudinary.com/wonderwoman/image/upload/v1665273260/3D_gold_crown_logo_-_Made_with_PosterMyWall_n1vleo.png"
        />
      </a>
    </div>
  </nav>,
]
`;
